src/Drush/Commands/InitializeBuildScriptsDrushCommands.php

```diff
@@ -0,0 +1,196 @@
+use DrupalCodeGenerator\Asset\AssetCollection;
+use Drush\Attributes as CLI;
+use Drush\Drush;
+
+/**
+ * Combawa initialize build scripts drush command.
+ */
+class InitializeBuildScriptsDrushCommands extends DrushCommandsGeneratorBase implements SiteAliasManagerAwareInterface {
+
+  use SiteAliasManagerAwareTrait;
+
+  const TEMPLATES_PATH = __DIR__ . '/../../../templates/combawa-build';
+
+  /**
+   * Generate environment command.
+   */
+  #[CLI\Command(name: 'combawa:initialize-build-scripts', aliases: ['ibs'])]
+  #[CLI\Option(
+    name: 'build-mode',
+    description: 'The build mode to use. (Accepted: <info>install</info>, <info>update</info>)',
+    suggestedValues: ['install', 'update'])]
+  #[CLI\Option(
+    name: 'overwrite-scripts',
+    description: 'Overwrite existing scripts files. (Accepted: <info>boolean</info>)',
+    suggestedValues: [TRUE, FALSE])]
+  #[CLI\Option(
+    name: 'generate-env',
+    description: 'Generate environment file. (Accepted: <info>boolean</info>)',
+    suggestedValues: [TRUE, FALSE])]
+  #[CLI\Option(
+    name: 'dry-run',
+    description: 'Output the generated code but do not save it to file system.')]
+  #[CLI\Usage(name: 'drush combawa:initialize-build-scripts', description: 'Run with wizard')]
+  public function generateEnvironment(array $options = [
+    'build-mode' => self::REQ,
+    'overwrite-scripts' => self::OPT,
+    'generate-env' => self::OPT,
+    'dry-run' => FALSE,
+  ]): int {
+    return $this->generate($options);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function extractOptions(array $options): array {
+    $vars = [
+      'build_mode' => $options['build-mode'],
+      'overwrite_scripts' => $options['overwrite-scripts'],
+      'generate_env' => $options['generate-env'],
+    ];
+    return array_filter($vars, fn ($value) => !\is_null($value));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function interview(array &$vars): void {
+    if (!isset($vars['build_mode'])) {
+      $composerData = Json::decode(file_get_contents($this->drupalFinder()->getComposerRoot() . '/composer.json'));
+      $defaultValue = $composerData['extra']['combawa']['build_mode'] ?? 'install';
+      $choices = ['install', 'update'];
+      $choice = $this->io()->choice(
+        'What is the build mode to use?',
+        $choices,
+        $defaultValue,
+      );
+      $vars['build_mode'] = $choices[$choice];
+    }
+
+    $scriptsDir = $this->drupalFinder()->getComposerRoot() . '/scripts/combawa';
+    if (
+      $this->fileSystem->exists($scriptsDir . '/' . $vars['build_mode'] . '.sh') &&
+      !isset($vars['overwrite_scripts'])
+    ) {
+      $defaultValue = FALSE;
+      $vars['overwrite_scripts'] = $this->io()->confirm(
+        'Do you want to overwrite your existing scripts located in the scripts/combawa directory?',
+        $defaultValue,
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function validateVars(array $vars): void {
```

Comment:
I don't know if it works for options instead of arguments. Plus, I have no idea how this is implemented.
For now I only converted our Console code to Drush using examples from DCG that has similar stuff. Maybe we could improve that in a follow up?

Reply:
After some investigation, options doesn't provide validation hooks.
See \Consolidation\AnnotatedCommand\Hooks\HookManager